### PR TITLE
Issue 20104 browser sort order

### LINKED ISSUE #20104: Subfolders in Site Browser are not sorted alphabetically if the folder Name is changed
As reported in https://dotcms.zendesk.com/agent/tickets/103795 
Reported in v5.3.8.4
Recreated in Demo v21.03 

Subfolders in Site Browser are not sorted alphabetically if the folder Name is changed.

If a folder's Title and Name are NOT the same, dotCMS uses the Title field to sort alphabetically.
This makes the folders appear out of order when using the Site Browser.

**NOTE: Root-level folders DO get sorted on Name, not Title. Issue ONLY occurs on sub-folders.**

**To Reproduce**
1. Create a folder at root level
2. Create a few sub folders where the Titles and Names agree
3. They appear to be sorted alphabetically sorted
4. Add another folder where the Title and Name are different

The folder is sorted by Title not Name

SCREENSHOT ONE - folders appear out of order ('x-ray' appears before 'barn'):

image

SCREENSHOT TWO - Folder detail of 'x-ray' showing the title is 'baby'

image

SCREENSHOT THREE - changing the title to 'x-ray'

image

SCREENSHOT FOUR - folder now appears in appropriate alphabetical sort order

image

### PR DESCRIPTION
**Pull request by erickgonzalez**
- Changes the default sort to `mod_date desc` rather than `mod_date asc` for content.
- Sort the folders by `name` as default.

### CODE REVIEW

**Comment by wezell** [comment]
Which is the new default - by `mod_date desc`?

**Comment by erickgonzalez** [comment]
@wezell yes, the default for Content is now `mod_date desc` and for Folders is now `name` not `title`

**Review by jdotcms — APPROVED** [review]
looks good, add some test

File: dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java [hunk 1]
```diff
@@ -508,15 +508,36 @@ public void testingDifferentBrowserAPIResults() throws Exception{
...
+     * Given Scenario: Create a parent folder and a couple of subfolders, get the content of the parent folder.
+     * ExpectedResult: The subfolders should be sort by name not by title.
+     *
+     */
+    @Test
+    public void test_getFolderContent_folderOrderedByName() throws Exception{
+        final Host site = new SiteDataGen().nextPersisted();
+
+        final Folder parentFolder = new FolderDataGen().site(site).nextPersisted();
+        //crear folder title y name `barn`
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java:523]: comments in english :P

Comment (wezell) [reply dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java]: sip!

**Review by dsilvam — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

**Pull request merged**